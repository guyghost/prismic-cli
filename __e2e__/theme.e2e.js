const path = require('path');
const fs = require('fs');
const { spawnSync } = require('child_process');
const {
  login,
  changeBase,
  deleteRepo,
  TMP_DIR,
  PRISMIC_BIN,
  rmdir,
  genRepoName,
  mkdir,
  RETRY_TIMES,
} = require('./utils');

describe('prismic theme [ --theme-url | --folder | --conf | --template ]', () => {
  jest.retryTimes(RETRY_TIMES);
  jest.setTimeout(300000);

  const repoName = genRepoName('cli-theme-test-two');
  const dir = path.resolve(TMP_DIR, 'theme-test');

  beforeAll(async () => {
    changeBase();
    login();
    await deleteRepo(repoName);
    return rmdir(dir, { recursive: true }).finally(() => mkdir(TMP_DIR, { recursive: true }));
  });


  it('should create a project using a theme', () => {
    const args = [
      'theme',
      '--theme-url', 'https://github.com/prismicio/nuxtjs-blog.git',
      '--conf', 'nuxt.config.js',
      '--domain', repoName,
      '--folder', dir,
      '--skip-install',
    ];

    const res = spawnSync(PRISMIC_BIN, args, { encoding: 'utf8' });
    expect(res.stdout).toBeTruthy();
    // expect(res.stdout).toMatchSnapshot();

    expect(fs.existsSync(dir)).toBe(true);
    expect(res.status).toBeFalsy();
  });
});
