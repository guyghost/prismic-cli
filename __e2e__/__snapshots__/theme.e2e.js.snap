// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prismic theme [ --theme-url | --folder | --conf | --template ] should create a project using a theme 1`] = `
"We are checking the theme integrity
CAREFUL, your current base is: https://wroom.io

Let's get to it!
An unexpected error occured
Error: Repository creation failed. Status: 500
Initialize local project
Running npm install...
Your project is ready, to proceed:

Go to the project folder : cd /Users/marc/Projects/prismic/prismic-cli/__tmp__/theme-test
"
`;
